File: Python/OneLang/Test/TestCase.py
from onelang_core import *

class TestCase:
    def __init__(self, name, action):
        self.name = name
        self.action = action

class SyncTestCase(TestCase):
    def __init__(self, name, sync_action):
        self.sync_action = sync_action
        super().__init__(name, None)
        self.action = lambda artifact_dir: self.execute(artifact_dir)
    
    def execute(self, artifact_dir):
        self.sync_action(artifact_dir)

File: PHP/src/OneLang/Test/TestCase.php
<?php

namespace OneLang\Test\TestCase;

interface ITestCollection {
    function getTestCases();
}

class TestCase {
    public $name;
    public $action;
    
    function __construct($name, $action) {
        $this->name = $name;
        $this->action = $action;
    }
}

class SyncTestCase extends TestCase {
    public $syncAction;
    
    function __construct($name, $syncAction) {
        parent::__construct($name, null);
        $this->syncAction = $syncAction;
        $this->action = function ($artifactDir) { return $this->execute($artifactDir); };
    }
    
    function execute($artifactDir) {
        call_user_func($this->syncAction, $artifactDir);
    }
}
